src/comb_filter.rs

```diff
@@ -0,0 +1,151 @@
+pub struct CombFilter {
+    filter_type: FilterType,
+    max_delay_samples: usize,
+    delay_line: Vec<Vec<f32>>,
+    gain: f32,
+    sample_rate_hz: f32,
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum FilterType {
+    FIR,
+    IIR,
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum FilterParam {
+    Gain,
+    Delay,
+}
+
+#[derive(Debug, Clone)]
+pub enum Error {
+    InvalidValue { param: FilterParam, value: f32 },
+}
+
+impl CombFilter {
+    pub fn new(filter_type: FilterType, max_delay_secs: f32, sample_rate_hz: f32, num_channels: usize) -> Self {
+        let max_delay_samples = (max_delay_secs * sample_rate_hz) as usize;
+        let delay_line = vec![vec![0.0; max_delay_samples]; num_channels];
+        let gain = 1.0;
+        CombFilter {
+            filter_type,
+            max_delay_samples,
+            delay_line,
+            gain,
+            sample_rate_hz,
+        }
+    }
+
+    pub fn reset(&mut self) {
+        for channel in self.delay_line.iter_mut() {
+            for sample in channel.iter_mut() {
+                *sample = 0.0;
+            }
```

Comment:
I see you reset the samples, but it would also be useful to reset the parameters that could be changed by set_param like gain and delay.